superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -1249,3 +1249,27 @@ export function createDatasource(vizOptions) {
       });
   };
 }
+
+export function createCtasDatasource(vizOptions) {
+  return dispatch => {
+    dispatch(createDatasourceStarted());
+    return SupersetClient.post({
+      endpoint: '/superset/sqllab_table_viz/',
+      postPayload: { data: vizOptions },
+    })
+      .then(({ json }) => {
+        dispatch(createDatasourceSuccess(json));
+
+        return Promise.resolve(json);
```

Comment:
Promise.resolve() is unnecessary when you're in a promise method.

```suggestion
        return json;
```